provider/openstack/networking.go

```diff
@@ -228,18 +237,20 @@ func getExternalNeutronNetworksByAZ(e *Environ, azName string) ([]string, error)
 		}
 	}
 	// Find all external networks in availability zone
-	networks, err := neutron.ListNetworksV2()
+	networks, err := neutron.ListNetworksV2(externalNetworkFilter())
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
 	netIds := make([]string, 0)
 	for _, network := range networks {
-		if network.External == true {
-			for _, netAZ := range network.AvailabilityZones {
-				if azName == netAZ {
-					netIds = append(netIds, network.Id)
-					break
-				}
+		// TODO (hml): OpenStack Compute and Network AZs have no direct relation,
```

Comment:
Ah, interesting. so is there any restriction on what networks a machine can be attached to, AZ-wise? I believe this method exists to find the subset of networks a machine can be attached to, based on AZ. In AWS/EC2, each subnet exists in one AZ only. You can only attach a machine/NIC to a subnet if the machine is in the same AZ.